client/src/main/java/com/netflix/conductor/client/automator/TaskPollExecutor.java

```diff
@@ -65,15 +66,21 @@
 
         this.executorService = Executors.newFixedThreadPool(threadCount,
             new BasicThreadFactory.Builder()
-                .namingPattern(workerNamePrefix)
-                .uncaughtExceptionHandler(uncaughtExceptionHandler)
-                .build());
+                    .namingPattern(workerNamePrefix)
+                    .uncaughtExceptionHandler(uncaughtExceptionHandler)
+                    .build());
 
         this.pollingSemaphore = new PollingSemaphore(threadCount);
     }
 
     void pollAndExecute(Worker worker) {
-        if (eurekaClient != null && !eurekaClient.getInstanceRemoteStatus().equals(InstanceStatus.UP)) {
+        Boolean discoveryOverride = Optional.ofNullable(PropertyFactory.getBoolean(worker.getTaskDefName(), OVERRIDE_DISCOVERY, false))
+                .orElseGet(() -> Optional.ofNullable(PropertyFactory.getBoolean(ALL_WORKERS, DOMAIN, null))
```

Comment:
Did you mean `PropertyFactory.getBoolean(ALL_WORKERS, OVERRIDE_DISCOVERY, false)` here?

```java
boolean discoveryOverride = Optional.ofNullable(PropertyFactory.getBoolean(worker.getTaskDefName(), OVERRIDE_DISCOVERY, null))
                .orElseGet(() -> Optional.ofNullable(PropertyFactory.getBoolean(ALL_WORKERS, OVERRIDE_DISCOVERY, false));
```

Reply:
@aravindanr This should be fixed.

Reply:
Doesn't the default for the first call be null for the `orElseGet()` to be invoked? If its set as `false`, the else block will not be reached.

`PropertyFactory.getBoolean(worker.getTaskDefName(), OVERRIDE_DISCOVERY, null)`

The code could be simplified like this,

```java
boolean discoveryOverride = Optional.ofNullable(PropertyFactory.getBoolean(worker.getTaskDefName(), OVERRIDE_DISCOVERY, null))
                .orElseGet(() -> Optional.ofNullable(PropertyFactory.getBoolean(ALL_WORKERS, OVERRIDE_DISCOVERY, false));
```

Reply:
Yes. I have fixed this. If we remov ethe orElseGet in the end, then we need to return Boolean on the second line and not an Optional.